tensorzero-core/src/endpoints/stored_inferences/v1/types.rs

```diff
@@ -0,0 +1,221 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::db::inferences::InferenceOutputSource;
+use crate::stored_inference::StoredInference;
+
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[cfg_attr(test, ts(export))]
+pub struct FloatMetricFilter {
+    pub metric_name: String,
+    pub value: f64,
+    pub comparison_operator: FloatComparisonOperator,
+}
+
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[cfg_attr(test, ts(export))]
+pub struct BooleanMetricFilter {
+    pub metric_name: String,
+    pub value: bool,
+}
+
+/// Filter by tag key-value pair.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[cfg_attr(test, ts(export))]
+pub struct TagFilter {
+    pub key: String,
+    pub value: String,
+    pub comparison_operator: TagComparisonOperator,
+}
+
+/// Filter by timestamp.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[cfg_attr(test, ts(export))]
+pub struct TimeFilter {
+    #[cfg_attr(test, ts(type = "Date"))]
+    pub time: DateTime<Utc>,
+    pub comparison_operator: TimeComparisonOperator,
+}
+
+/// Comparison operators for float metrics.
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+#[cfg_attr(test, ts(export))]
+pub enum FloatComparisonOperator {
+    #[serde(rename = "<")]
+    LessThan,
+    #[serde(rename = "<=")]
+    LessThanOrEqual,
+    #[serde(rename = "=")]
+    Equal,
+    #[serde(rename = ">")]
+    GreaterThan,
+    #[serde(rename = ">=")]
+    GreaterThanOrEqual,
+    #[serde(rename = "!=")]
+    NotEqual,
+}
+
+/// Comparison operators for timestamps.
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+#[cfg_attr(test, ts(export))]
+pub enum TimeComparisonOperator {
+    #[serde(rename = "<")]
+    LessThan,
+    #[serde(rename = "<=")]
+    LessThanOrEqual,
+    #[serde(rename = "=")]
+    Equal,
+    #[serde(rename = ">")]
+    GreaterThan,
+    #[serde(rename = ">=")]
+    GreaterThanOrEqual,
+    #[serde(rename = "!=")]
+    NotEqual,
+}
+
+/// Comparison operators for tag filters.
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Serialize)]
+#[cfg_attr(test, ts(export))]
+pub enum TagComparisonOperator {
+    #[serde(rename = "=")]
+    Equal,
+    #[serde(rename = "!=")]
+    NotEqual,
+}
+
+/// The ordering direction.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[cfg_attr(test, ts(export))]
+pub enum OrderDirection {
+    #[serde(rename = "ascending")]
```

Comment:
I renamed these from `ASC` to `ascending` since nowhere else in the codebase do we expose APIs that look like SQL.